### PR TITLE
Updated LBANN to include pytest

### PR DESCRIPTION
**Pull request by bvanessen**
Updated LBANN to include pytest.  Added support for new versions of cudnn.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cudnn/package.py [hunk 1]
```diff
@@ -13,6 +13,10 @@ class Cudnn(Package):
 
     homepage = "https://developer.nvidia.com/cudnn"
 
+    version('7.6.3-10.1-x86_64', '463a4258c007599b42f04f5008116e40',
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/cudnn/package.py]: Hi @bvanessen the checksums are now required to be sha256 vs. md5. Could you update them?

File: var/spack/repos/builtin/packages/lbann/package.py [hunk 2]
```diff
@@ -112,6 +112,8 @@ class Lbann(CMakePackage):
     depends_on('py-onnx@1.3.0:', type='run', when='@:0.90,0.99:')
     depends_on('py-pandas@0.24.1:', type='run', when='@:0.90,0.99:')
     depends_on('py-texttable@1.4.0:', type='run', when='@:0.90,0.99:')
+    depends_on('py-pytest', type='run', when='@:0.90,0.99:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lbann/package.py]: Usually pytest is a `type='test'` dependency. How is it used at run-time in this package?

**Comment by bvanessen** [comment]
I addressed the comments.

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**